hdk-rust/src/api.rs

```diff
@@ -376,3 +406,680 @@ pub fn close_bundle(_action: BundleOnClose) -> Result<(), RibosomeError> {
     // FIXME
     Err(RibosomeError::FunctionNotImplemented)
 }
+//--------------------------------------------------------------------------------------------------
+// UNIT TESTS
+//--------------------------------------------------------------------------------------------------
+
+/// Unit tests
+#[cfg(test)]
+mod test {
```

Comment:
Ah! All those tests below don't run because it's `mod test` instead of `mod tests`.

Reply:
@lucksus, I'd thought `mod test` was the Rust convention, although any function (whatever its name) annotated with `[#cfg(test)]` should be run by the cargo test harness. I'd successfully run these unit tests via `cargo test` and `make test` natively on Debian but I've since run them again via `. docker/docker-test` and they run fine (subject to the reported failures as noted above). Is there something special about the Travis-based(?) CI environment?

Reply:
you can put `#[cfg(test)]` on anything to elide a code block whenever the tests for its crate are not currently being run

`#[test]` flags a function as a test to be run

Reply:
Thanks @thedavidmeister. Will revise to disable running unit tests pending implementation of functions under test; i.e.:
```rust
/// #[test] // FIXME when implemented
```

Reply:
Unfortunately, this approach I described above results in unreferenced functions so I've reverted to my previous suggestion of only enabling those unit tests that pass and commenting out those that crash or are otherwise pending implementation of the function(s) under test.